### PR TITLE
func: added should delete orginal after crop functionality

### LINKED ISSUE #11800: Error after crop and save to gallery
### The bug

After cropping a photo and trying to save it to the gallery:
Screenshot_20240815_103146

### The OS that Immich Server is running on

Unraid

### Version of Immich Server

1.112.0-1

### Version of Immich Mobile App

1.112.1 b154

### Platform with the issue

- [ ] Server
- [ ] Web
- [X] Mobile

### Your docker-compose.yml content

```YAML
.
```

### Your .env content

```Shell
.
```

### Reproduction steps

```bash
1. Crop photo,
2. Save to gallery
3. Error: Null check operator used on a null value
```

### Relevant log output

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by yashrajjain726**
Fixed #11800
- Added new functionality of asking user to delete orginal image after cropping the image.

### CODE REVIEW

**Comment by yashrajjain726** [comment]
@mertalev Please add someone to review

**Comment by mertalev** [comment]
The linked issue doesn't seem related to this PR. Also, the edited image is lossy and loses some qualities of the image, like HDR. I think we should make that more robust before suggesting the user to delete the original.

**Comment by yashrajjain726** [comment]
The cropping functionality is acting the same as previously, as I have not changed anything regarding the cropping functionality.

I have just added the "delete the original" functionality after the cropping is done. For, cropping image quality functionality, you can create a new bug card, and I can start working on it.

Also this functionality has nothing to do, within cropping functionality, so this just acts as a adds on as a functionality over the cropping functionality.

**Comment by mertalev** [comment]
You haven't changed it, sure. But the premise of the feature is that the user can effectively replace the original with the edited version. I don't think we should suggest this until the edited image has the same quality as the original.

**Comment by yashrajjain726** [comment]
Ok, understood. I hope we can merge this after the crop image quality fix, as you mentioned. Please create a new issue regarding that, and let me know, I can start on that.

File: mobile/assets/i18n/en-US.json [hunk 1]
```diff
@@ -600,5 +600,7 @@
   "download_paused": "Download paused",
   "download_enqueue": "Download enqueued",
   "download_notfound": "Download not found",
-  "download_waiting_to_retry": "Waiting to retry"
+  "download_waiting_to_retry": "Waiting to retry",
+  "crop_page_delete_orginal_title": "DDelete Original?",
```
Comment (Clement83) [inline mobile/assets/i18n/en-US.json:604]: DDelete Original ? Double D on Delete

Comment (yashrajjain726) [reply mobile/assets/i18n/en-US.json]: Sorry, my bad. Updated it in the new commit.

**Comment by yashrajjain726** [comment]
Any update on this @Clement83

**Comment by yashrajjain726** [comment]
Great @Clement83 , As it was assigned to you, I thought you would be reviewer.

**Comment by yashrajjain726** [comment]
@mertalev  Any update on this ?

**Comment by mertalev** [comment]
Sorry, I'm not a mobile dev and won't be the best person to review the implementation details.

My suggestion is to split out the lossless conversion to its own PR. In particular, it'd be nice to compare the output images before and after the change, and to check its color processing, metadata, etc.

**Comment by alextran1502** [comment]
Thanks for the PR. However, this is not the functionality I envision in the app. For future reference, you can chat with us on Discord before opening a PR to avoid situations where work got put in but doesn't get merged. Again, sorry for the confusion.

**Comment by yashrajjain726** [comment]
Great, not an issue. Can we close #11800 also, if not required ?

**Pull request closed without merging**